packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -274,7 +278,7 @@ export function useAsyncData<
 
   // Allow directly awaiting on asyncData
   const asyncDataPromise = Promise.resolve(nuxt._asyncDataPromises[key]).then(() => asyncData) as AsyncData<ResT, DataE>
```

Comment:
I believe it should be something like:
```suggestion
  const asyncDataPromise = options.strategy === 'blocking'
    ? Promise.resolve(nuxt._asyncDataPromises[key]).then(() => asyncData) as AsyncData<ResT, DataE>
    : Promise.resolve(asyncData)
```
right?
Then, below line could be restored to `Object.assign(asyncDataPromise, asyncData)`

The `Object.assign` in current form seems to be doing nothing for strategies different than `blocking`. It copies `asyncData` **into** `Promise.resolve(asynData)` object, which is not used anywhere later on.